### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by sbruinsje** [Comcast/react-data-grid#3581] [Q&A]
Does the main readme of this repo contain the entire and only documentation? I recently discovered this project but I am really confused on where to find up-to-date and complete documentation. I spent hours trying to find out how to customize the rendering of column headers and the only way I found out is that I happened to stumble upon a demo that used `renderHeaderCell` in the source code.

The previous documentation site looked much clearer but its not up-to-date. Any plans on bringing something similar back? Or am I looking in the wrong place?

**Comment by nihilum-labs** [comment]
I use AI in such cases. Usually, if documentation is lacking, ChatGPT/Claude helps a lot. This, of course, does not cancel the desire to have cool, clear and complete documentation for such a cool project as react-data-grid.

**Comment by Ben-CA** [comment]
This seems to be a powerful grid, and I like that it's open source and seemingly critical features aren't limited to the (expensive) paid versions. But I've found the documentation lacking as well. I think it would get a lot more use if the documentation was better. The extent seems to be pretty much:

**There's the demo:**
https://adazzle.github.io/react-data-grid/#/AllFeatures
**With the code then being:**
https://github.com/adazzle/react-data-grid/blob/main/website/routes/AllFeatures.lazy.tsx

I've tried several AI models, and specifically tell it for react-data-grid, but I find about 75% of the time the AI suggests things that don't work for react-data-grid. (It pulls the suggestions instead from things you can do with AG Grid, MUI X-data-grid, etc.) Sometimes it gives clues, but you end up spending a lot more time trying things and debugging.
